### PR TITLE
Upgrade third party CLI11 version

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

Third party dependency upgrade

# Motivation

The aim is to use it as our CLI tool (instead of getopt).

### CODE REVIEW

**Review by nsavoire — DISMISSED** [review]
LGTM !

**Comment by r1viollet** [comment]
Really thankful that the CI is now failing in Github :+1: (I might not have expected this conversion issue)

File: test/simple_malloc.cc [hunk 1]
```diff
@@ -177,99 +177,96 @@ int main(int argc, char *argv[]) {
...
-    }
-    wrapper(opts, stats[0]);
-    for (auto &t : threads) {
-      t.join();
-    }
-    auto pid = getpid();
-    for (auto &stat : stats) {
-      print_stats(pid, stat);
-    }
-  } catch (...) { return 1; }
```
Comment (r1viollet) [inline test/simple_malloc.cc:274]: @nsavoire I did not see the point of the try / catch (I was actually confused as to why this was failing), are you OK removing this ?

Comment (nsavoire) [reply test/simple_malloc.cc:274]: I think it's a good practice to catch all exceptions in main, here what was missing was printing errors I guess.
What was the error ?

**Review by nsavoire — APPROVED** [review]
LGTM !

File: test/simple_malloc.cc (hunk 1 above)
Comment (r1viollet) [reply test/simple_malloc.cc:274]: As discussed, leaving exception propagate, makes the crash more explicit.

**Pull request merged**